File: csharp/allergies/Allergies.cs
using System;
using System.Linq;

public enum Allergen
{
    Eggs,
    Peanuts,
    Shellfish,
    Strawberries,
    Tomatoes,
    Chocolate,
    Pollen,
    Cats
}

public class Allergies
{
    private int mask;

    public Allergies(int mask)
    {
        this.mask = mask;
    }

    public bool IsAllergicTo(Allergen allergen) => (1 << (int)allergen & mask) != 0;

    public Allergen[] List() =>
        Enum.GetValues(typeof(Allergen))
            .OfType<Allergen>()
            .Where(IsAllergicTo)
            .ToArray();
}

File: swift/allergies/Sources/Allergies/Allergies.swift
enum Allergen: Int {
    case eggs, peanuts, shellfish, strawberries, tomatoes, chocolate, pollen, cats
}

struct Allergies {
    let mask: UInt
    
    init(_ mask: UInt) {
        self.mask = mask
    }
    
    func hasAllergy(_ allergen: Allergen) -> Bool {
        (1 << allergen.rawValue) & self.mask != 0
    }
}
